vortex-expr/src/pruning/pruning_expr.rs

```diff
@@ -120,13 +79,37 @@ pub fn pruning_expr(expr: &ExprRef) -> Option<(ExprRef, Relation<AccessPath, Sta
     Some((expr, relation))
 }
 
+// Build a pruning expr mask an existing bundle of stats
+/// Create a stat based falsification expr using the stats in the `scope_field_paths`.
+/// These are of the form
+/// [["col_0", ..., "col_n", "stat_name"], ...] for each stat.
+pub fn checked_pruning_expr(
+    expr: &ExprRef,
+    scope_field_paths: &ScopeFieldPathSet,
+) -> Option<(ExprRef, RequiredStats)> {
+    let mut catalog = ScopeStatsCatalog {
+        any_catalog: Default::default(),
+        scope_field_paths,
+    };
+
+    let expr = expr.stat_falsification(&mut catalog)?;
+
+    // TODO: filter by used exprs
```

Comment:
Why is this TODO? Is that not what lines 98-101 are doing?

Reply:
Nope, there is no filtering by expr being in the falsification expr just yet